### ISSUE TITLE
[Bug] About switch

### ISSUE THREAD

**Issue by MuyueJusan** [zencq/NomNom#134]
### Description
I edited the save file for my switch version of the game using nomnom, but when I tried to import it back to the switch, the save file was unreadable.
### Reproduce
1. Go to 'file'
2. Click on 'open'
3. open my switch save
4. modify units

### Expected Behavior
The game archive is read normally
### Files
[save.zip](https://github.com/zencq/NomNom/files/12564650/save.zip)

### Additional Context
Add any other context about the problem here or at least complete the following information.
 - Version: [4.30.0.33-beta.3]
 - Platform: [switch]

**Comment by cengelha** [comment]
Thanks for the save! I found the problem and the next version should work again :)

**Comment by MuyueJusan** [comment]
> Thanks for the save! I found the problem and the next version should work again :)

I have seen this problem appear earlier, but no one has provided the game progress file and assisted in testing. If you need my help, I can help test

**Comment by MuyueJusan** [comment]
Thank you @cengelha . 
I have seen this problem appear earlier, but no one has provided the game progress file and assisted in testing. If you need my help, I can help test

**Comment by MuyueJusan** [comment]
@cengelha The problem I encountered may be the same as https://github.com/zencq/NomNom/issues/105

**Comment by cengelha** [comment]
@MuyueJusan Yeah... So, your save came just in time, that I could check it before the next release 👍 

How can I reach you to send you a dev version you can try? It's too big to attach here...

**Comment by MuyueJusan** [comment]
@cengelha  You can add me as a Discord friend or a Telegram contact, my Discord username is Hayai-Tsuki#5278, my Telegram link is https://t.me/muyueshisan

**Comment by MuyueJusan** [comment]
@cengelha You can add me as a Discord friend or a Telegram contact, my Discord username is **Hayai-Tsuki#5278**, my Telegram link is **https://t.me/muyueshisan**

**Comment by MuyueJusan** [comment]
@cengelha You can add me as a Discord friend or a Telegram contact, my Discord username is Hayai-Tsuki#5278, my Telegram link is https://t.me/muyueshisan

**Comment by MuyueJusan** [comment]
@cengelha I joined your Discord server, I should be able to contact you soon  :)
